floor_generator/test/processor/view_processor_test.dart

```diff
@@ -0,0 +1,274 @@
+      final classElement = await _createClassElement('''
+      @DatabaseView("SELECT * from otherentity")
+      class Person {
+        final int id;
+      
+        final String name;
+        
+        final String bar;
+      
+        Person({this.id, this.name, this.bar});
+      }
+    ''');
+
+      final actual = ViewProcessor(classElement).process().constructor;
+
+      const expected =
+          "Person(id: row['id'] as int, name: row['name'] as String, bar: row['bar'] as String)";
+      expect(actual, equals(expected));
+    });
+
+    test('generate constructor with optional argument', () async {
+      final classElement = await _createClassElement('''
+      @DatabaseView("SELECT * from otherentity")
+      class Person {
+        final int id;
+      
+        final String name;
+        
+        final String bar;
+      
+        Person(this.id, this.name, [this.bar]);
+      }
+    ''');
+
+      final actual = ViewProcessor(classElement).process().constructor;
+
+      const expected =
+          "Person(row['id'] as int, row['name'] as String, row['bar'] as String)";
+      expect(actual, equals(expected));
+    });
+
+    test('generate constructor with optional arguments', () async {
+      final classElement = await _createClassElement('''
+      @DatabaseView("SELECT * from otherentity")
+      class Person {
+        final int id;
+      
+        final String name;
+        
+        final String bar;
+      
+        Person([this.id, this.name, this.bar]);
+      }
+    ''');
+
+      final actual = ViewProcessor(classElement).process().constructor;
+
+      const expected =
+          "Person(row['id'] as int, row['name'] as String, row['bar'] as String)";
+      expect(actual, equals(expected));
+    });
+  });
+
+  group('@Ignore', () {
+    test('ignore field not present in constructor', () async {
+      final classElement = await _createClassElement('''
+      @DatabaseView("SELECT * from otherentity")
+      class Person {
+        final int id;
+      
+        final String name;
+        
+        @ignore
+        String foo;
+      
+        Person(this.id, this.name);
+      }
+    ''');
+
+      final actual = ViewProcessor(classElement)
+          .process()
+          .fields
+          .map((field) => field.name);
+
+      const expected = 'foo';
+      expect(actual, isNot(contains(expected)));
+    });
+
+    test('ignore field present in constructor', () async {
+      final classElement = await _createClassElement('''
+      @DatabaseView("SELECT * from otherentity")
+      class Person {
+        final int id;
+      
+        final String name;
+        
+        @ignore
+        String foo;
+      
+        Person(this.id, this.name, [this.foo = 'foo']);
+      }
+    ''');
+
+      final actual = ViewProcessor(classElement).process().constructor;
+
+      const expected = "Person(row['id'] as int, row['name'] as String)";
+      expect(actual, equals(expected));
+    });
```

Comment:
These are basically duplicates from `entity_processor_test`. After migrating duplicate code to `QueryableProcessor`, we could only test this functionality there which would allow us to get rid of this duplication.

Reply:
I don't see a simple way to achieve this, tbh, since `QueryableProcessor` is an abstract class. We would have to instantiate the concrete types of `ViewProcessor` and `EntityProcessor` anyway... But if you have a concrete vision on how to do it, please share and I will try.

Reply:
We could just create an implementation class of `QueryableProcessor` in the test and asserts its methods for correctness.

```dart
class UnderTest extends QueryableProcessor<Object> {
  UnderTest(ClassElement classElement) : super(classElement);

  @override
  Object process() {
    return null;
  }
}
```

Reply:
I'll try!